dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java

```diff
@@ -123,19 +130,37 @@ private void successHandler(Response response) {
      * Failure handler.
      *
      * @param logMessage the log message
+     * @param shouldFailOnErrors should fail on error
      */
-    public void failureHandler(String logMessage) {
+    public void failureHandler(String logMessage, boolean shouldFailOnErrors) {
         postResponseTelemetry.sendFailureTelemetry(meterStatsManager, startTime);
         LOGGER.error(logMessage);
         Exception httpFailureException = new HttpFailureException(logMessage);
-        if (httpSourceConfig.isFailOnErrors()) {
+        if (shouldFailOnErrors) {
             reportAndThrowError(httpFailureException);
         } else {
             errorReporter.reportNonFatalException(httpFailureException);
         }
         resultFuture.complete(Collections.singleton(rowManager.getAll()));
     }
 
+    private boolean shouldFailOnError(Integer statusCode) {
+        if (statusCode == 0 || StringUtil.isNullOrEmpty(httpSourceConfig.getExcludeFailOnErrorsCodeRange())) {
+            return true;
+        }
+        return !getExcludeFailOnErrorCodes(httpSourceConfig.getExcludeFailOnErrorsCodeRange()).contains(statusCode);
+    }
+
+    private HashSet<Integer> getExcludeFailOnErrorCodes(String input) {
```

Comment:
We should save the list of error codes and not generate for every response. the list can be created and saved outside of the Response Handler.

Reply:
agree, made changes